components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/internal/AuthenticationDataPublisherServiceComponent.java

```diff
@@ -18,97 +18,137 @@
-import org.wso2.carbon.identity.data.publisher.application.authentication.impl.DASLoginDataPublisherImpl;
-import org.wso2.carbon.identity.data.publisher.application.authentication.impl.DASSessionDataPublisherImpl;
+import org.wso2.carbon.identity.event.services.IdentityEventService;
 import org.wso2.carbon.registry.core.service.RegistryService;
 import org.wso2.carbon.user.core.service.RealmService;
 
-import java.util.Collections;
-
-/**
- * @scr.component name="org.wso2.carbon.identity.data.publisher.authn" immediate="true"
- * @scr.reference name="registry.service"
- * interface="org.wso2.carbon.registry.core.service.RegistryService"
- * cardinality="1..1" policy="dynamic" bind="setRegistryService"
- * unbind="unsetRegistryService"
- * @scr.reference name="realm.service" interface="org.wso2.carbon.user.core.service.RealmService"
- * cardinality="1..1" policy="dynamic" bind="setRealmService"
- * unbind="unsetRealmService"
- * @scr.reference name="eventStreamManager.service"
- * interface="org.wso2.carbon.event.stream.core.EventStreamService" cardinality="1..1"
- * policy="dynamic" bind="setEventStreamService" unbind="unsetEventStreamService"
- * unbind="unsetEventStreamService"
- * @scr.reference name="identity.authentication.data.publisher"
- * interface="org.wso2.carbon.identity.application.authentication.framework.AuthenticationDataPublisher"
- * cardinality="0..n" policy="dynamic" bind="setAuthenticationDataPublisher"
- * unbind="unsetAuthenticationDataPublisher"
- */
+@Component(
+        name = "identity.data.publisher.authn",
+        immediate = true
+)
 public class AuthenticationDataPublisherServiceComponent {
 
-    protected void activate(ComponentContext context) {
+    private static Log log = LogFactory.getLog(AuthenticationDataPublisherServiceComponent.class);
 
-        BundleContext bundleContext = context.getBundleContext();
-        bundleContext
-                .registerService(AuthenticationDataPublisher.class.getName(), new DASLoginDataPublisherImpl(), null);
-        bundleContext
-                .registerService(AuthenticationDataPublisher.class.getName(), new DASSessionDataPublisherImpl(), null);
-        bundleContext
-                .registerService(AuthenticationDataPublisher.class.getName(), new AuthenticationAuditLogger(), null);
-        bundleContext
-                .registerService(AuthenticationDataPublisher.class.getName(), new AuthnDataPublisherProxy(), null);
+    @Activate
+    protected void activate(ComponentContext context) {
 
+        try {
+            BundleContext bundleContext = context.getBundleContext();
+            bundleContext
+                    .registerService(AuthenticationDataPublisher.class.getName(), new AuthnDataPublisherProxy(), null);
+            if (log.isDebugEnabled()) {
+                log.debug("org.wso2.carbon.identity.data.publisher.application.authentication bundle is activated");
+            }
+        } catch (Exception e) {
```

Comment:
Usually we catch throwable here to make sure no failures take place.

Reply:
Changed to throwable.